src/commands/keygen.rs

```diff
@@ -0,0 +1,605 @@
+  -f		force override of existing symlinks
+  -v		show the version and exit
+  The following files will be created:
+    K<name>+<alg>+<id>.key	Public key in RR format
+    K<name>+<alg>+<id>.private	Private key in key format
+    K<name>+<alg>+<id>.ds	DS in RR format (only for DNSSEC KSK keys)
+  The base name (K<name>+<alg>+<id>) will be printed to stdout
+";
+
+impl LdnsCommand for Keygen {
+    const HELP: &'static str = LDNS_HELP;
+
+    fn parse_ldns<I: IntoIterator<Item = OsString>>(args: I) -> Result<Self, Error> {
+        let mut algorithm = None;
+        let mut make_ksk = false;
+        let mut bits = 2048;
+        let mut create_symlinks = false;
+        let mut force_symlinks = false;
+        let mut name = None;
+
+        let mut parser = lexopt::Parser::from_args(args);
+
+        while let Some(arg) = parser.next()? {
+            match arg {
+                Arg::Short('a') => {
+                    if algorithm.is_some() {
+                        return Err("cannot specify algorithm (-a) more than once".into());
+                    }
+
+                    algorithm = parse_os_with("algorithm (-a)", &parser.value()?, |s| {
+                        Ok(match s {
+                            "list" => {
+                                // TODO: Mock stdout and process exit?
+                                println!("Possible algorithms:");
+                                println!("  - RSASHA256 (8)");
+                                println!("  - ECDSAP256SHA256 (13)");
+                                println!("  - ECDSAP384SHA384 (14)");
+                                println!("  - ED25519 (15)");
+                                println!("  - ED448 (16)");
+                                std::process::exit(0);
+                            }
+
+                            "RSASHA256" | "8" => Some(SecAlg::RSASHA256),
+                            "ECDSAP256SHA256" | "13" => Some(SecAlg::ECDSAP256SHA256),
+                            "ECDSAP384SHA384" | "14" => Some(SecAlg::ECDSAP384SHA384),
+                            "ED25519" | "15" => Some(SecAlg::ED25519),
+                            "ED448" | "16" => Some(SecAlg::ED448),
+
+                            _ => {
+                                return Err(format!("Invalid value {s:?} for algorithm (-a)"));
+                            }
+                        })
+                    })?;
+                }
+
+                Arg::Short('k') => {
+                    // NOTE: '-k' can be repeated, to no effect.
+                    make_ksk = true;
+                }
+
+                Arg::Short('b') => {
+                    // NOTE: '-b' can be repeated; the last instance wins.
+                    bits = parse_os("bits (-b)", &parser.value()?)?;
+                }
+
+                Arg::Short('r') => {
+                    // NOTE: '-r' can be repeated; we don't use it, so the order doesn't matter.
+                    let _ = parser.value()?;
```

Comment:
Do we want to warn or error here? Sounds like something that could lead to unexpected behaviour.

Reply:
Do you mean warning that we don't support `-r` at all?  Perhaps that's warranted, yeah.

Reply:
Explicit error saying that it's not supported yeah.